experimental/packages/opentelemetry-sdk-metrics-base/src/MeterProvider.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import * as metrics from '@opentelemetry/api-metrics';
+import { Resource } from '@opentelemetry/resources';
+import { Measurement } from './Measurement';
+import { Meter } from './Meter';
+import { MetricExporter } from './MetricExporter';
+import { MetricReader } from './MetricReader';
+import { View } from './View';
+
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#meterprovider
+
+export type MeterProviderOptions = {
+    resource?: Resource;
+}
+
+export class MeterProvider {
+    private _resource: Resource;
+    private _shutdown = false;
+    private _metricReaders: MetricReader[] = [];
+    private _metricExporters: MetricExporter[] = [];
+    private _views: View[] = [];
+
+    constructor(options: MeterProviderOptions) {
+        this._resource = options.resource ?? Resource.empty();
+    }
+
+    /**
+     * **Unstable**
+     * 
+     * This method is only here to prevent typescript from complaining and may be removed.
+     */
+    getResource() {
+        return this._resource;
+    }
+
+    getMeter(name: string, version = '', options: metrics.MeterOptions = {}): metrics.Meter {
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#meter-creation
+        if (this._shutdown) {
+            api.diag.warn('A shutdown MeterProvider cannot provide a Meter')
+            return metrics.NOOP_METER;
+        }
+
+        // Spec leaves it unspecified if creating a meter with duplicate
+        // name/version returns the same meter. We create a new one here
+        // for simplicity. This may change in the future.
+        return new Meter(this, { name, version }, options.schemaUrl);
```

Comment:
i think we should match the behavior of the trace sdk (https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/src/BasicTracerProvider.ts#L94), so maybe replace this by a TODO ?

Reply:
Added TODO